### PR TITLE
Web server integration test + api small refactor

### PR DESCRIPTION
**Pull request by iulianpascalau**
- added web server test processor node
- API small refactor to apply Dependency Inversion Principle
- add integration test that prove backwards compatibility on gas cost route

### CODE REVIEW

**Review by SebastianMarian — DISMISSED** [review]

File: integrationTests/interface.go [hunk 1]
```diff
@@ -39,3 +50,42 @@ type NetworkShardingUpdater interface {
 	UpdatePeerIdShardId(pid core.PeerID, shardId uint32)
 	IsInterfaceNil() bool
 }
+type (
```
Comment (sasurobert) [inline integrationTests/interface.go:53]: add empty line before

Comment (sasurobert) [reply integrationTests/interface.go:53]: why is type "(" needed ? couldn't be type Facade interface { etc }

Comment (iulianpascalau) [reply integrationTests/interface.go:53]: Didn't knew this was even possible. Changing to align with the rest of the code.

Comment (iulianpascalau) [reply integrationTests/interface.go]: Done

**Review by SebastianMarian — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

**Review by valentincraciun99 — APPROVED** [review]

**Pull request merged**